### ISSUE TITLE
Fix uses of `setDaemon`

### ISSUE THREAD

**Issue by mdickinson** [enthought/traitsui#1576]
There are a couple of uses of `thread.setDaemon(True)` in demo code. The `setDaemon` method [is deprecated](https://bugs.python.org/issue43723) in Python 3.10; we should replace `thread.setDaemon(True)` with `thread.daemon = True`.

```
mdickinson@mirzakhani traitsui % grin setDaemon
./traitsui/examples/demo/Advanced/Property_List_demo.py:
  122 :         thread.setDaemon(True)
./traitsui/examples/demo/Advanced/Statusbar_demo.py:
   72 :         thread.setDaemon(True)
```

**Comment by rahulporuri** [comment]
it looks like the `daemon` property is available in python 3.6 - https://docs.python.org/3.6/library/threading.html#threading.Thread.daemon

**Comment by mdickinson** [comment]
> it looks like the daemon property is available in python 3.6

Yes. The non-camel-case versions of the names go back a _long_ way, to Python 2.6.
